### PR TITLE
Scenario: 'Visitor can navigate the site'

### PR DESCRIPTION
**Pull request by raptorf1**
[PT Story](https://www.pivotaltracker.com/story/show/165594709)

# User Story
```
As a visitor,
In order to navigate the site
I would like to use a navbar
```

### What we worked on in this feature
* Adds feature test with 10 steps
* Creates 2 controllers :products & :about_us
* Creates a separate HAML file for the NavBar and renders it to the application view
* Adds the Tailwind CSS framework for the whole application

### What we learned
* Git-ponging between 3 people
* Connection between views (rendering the _navigation partial file)
* HAML syntax

### Screenshots
Screenshot from 2019-05-02 18-47-21

Screenshot from 2019-05-02 18-47-24

### CODE REVIEW

File: app/controllers/about_us_controller.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class AboutUsController < ApplicationController
+  def index
+  end
+end
```
Comment (faraznaeem) [inline app/controllers/about_us_controller.rb]: Why is this controller needed?

File: config/routes.rb [hunk 2]
```diff
@@ -1,3 +1,6 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  get 'about_us/index'
+  get 'products/index'
+
+  root "products#index"
```
Comment (faraznaeem) [inline config/routes.rb]: This is not how we write routes

File: features/visitor_can_navigate_the_site.feature [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        And I should see "MENU"
+        When I click "MENU" link
+        Then I should be on the landing page
+
+
+        
+
+
+    
+
```
Comment (faraznaeem) [inline features/visitor_can_navigate_the_site.feature]: Delete empty lines

**Review by faraznaeem — CHANGES_REQUESTED** [review]

**Comment by faraznaeem** [comment]
Please add a screenshot as well

File: features/visitor_can_navigate_the_site.feature (hunk 3 above)
Comment (oliverochman) [inline features/visitor_can_navigate_the_site.feature]: remove empty lines

File: features/step_definitions/basic_steps.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+Given("I visit the landing page") do
+  visit root_path
+end
+  
+Then("I should see {string}") do |content|
```
Comment (oliverochman) [inline features/step_definitions/basic_steps.rb]: add this step to the assertions steps instead

File: features/step_definitions/basic_steps.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+  expect(page).to have_content content 
+end
+  
+Then("I click {string} link") do |link|
```
Comment (oliverochman) [inline features/step_definitions/basic_steps.rb]: add this step to the assertions steps instead

File: features/visitor_can_navigate_the_site.feature [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    In order to navigate the site
+    I would like to use a navbar
+
+    Scenario:
+        Given I visit the landing page
+        Then I should see "ABOUT US"
+        And I should see "MENU"
+        And I should see "Opening hours"
+        And I should see "Park location"
+        Then I click "ABOUT US" link
```
Comment (oliverochman) [inline features/visitor_can_navigate_the_site.feature]: This should be "When"

File: config/routes.rb [hunk 7]
```diff
@@ -1,3 +1,6 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  get 'about_us/index'
```
Comment (oliverochman) [inline config/routes.rb]: update the way you define these routes

e..g.
`root controller: :products, action: :index`
`resources :about_us, only: [:index]`

File: app/controllers/about_us_controller.rb [hunk 8]
```diff
@@ -0,0 +1,4 @@
+class AboutUsController < ApplicationController
```
Comment (oliverochman) [inline app/controllers/about_us_controller.rb]: I would say that adding an about page in this feature is out of scope

File: app/views/layouts/application.html.haml [hunk 9]
```diff
@@ -2,9 +2,11 @@
 %html
   %head
     %meta{content: 'text/html; charset=UTF-8', http: {equiv: "Content-Type"}}/
-    %title Boilerplate
+    %title Kimchen FoodTruck
     = csrf_meta_tags
     = stylesheet_link_tag    'application', media: 'all', data: {'turbolinks-track' => 'reload'}
     = javascript_include_tag 'application', data: {'turbolinks-track' => 'reload'}
+    %link(href="https://cdn.jsdelivr.net/npm/tailwindcss/dist/tailwind.min.css" rel="stylesheet")
```
Comment (oliverochman) [inline app/views/layouts/application.html.haml]: Adding tailwind in this story is out of scope

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: app/controllers/about_us_controller.rb (hunk 8 above)
Comment (Carrosen) [reply app/controllers/about_us_controller.rb]: Okay, but without navigation, why is there a feature about this? Since it's about navigation. I don't really get what this feature is without a second page?

File: config/routes.rb (hunk 2 above)
Comment (Carrosen) [reply config/routes.rb]: true

File: features/visitor_can_navigate_the_site.feature (hunk 3 above)
Comment (Carrosen) [reply features/visitor_can_navigate_the_site.feature]: on it

Comment (Carrosen) [reply features/visitor_can_navigate_the_site.feature]: on it

File: features/step_definitions/basic_steps.rb (hunk 4 above)
Comment (Carrosen) [reply features/step_definitions/basic_steps.rb]: on it

File: features/step_definitions/basic_steps.rb (hunk 5 above)
Comment (Carrosen) [reply features/step_definitions/basic_steps.rb]: on it

File: app/views/layouts/application.html.haml (hunk 9 above)
Comment (Carrosen) [reply app/views/layouts/application.html.haml]: true

File: app/controllers/about_us_controller.rb (hunk 1 above)
Comment (raptorf1) [reply app/controllers/about_us_controller.rb]: The way we designed the LO-FI's we have an 'About Us' section we would like to use for the history let's say of the restaurant.
Since this feature is about navigating the site we wrote steps that require the user to click on the 'About Us' link and back.
To have an 'About Us' view then we must have an appropriate controller. Simple one with index only. So as to navigate to the page.

File: app/views/layouts/application.html.haml (hunk 9 above)
Comment (Carrosen) [reply app/views/layouts/application.html.haml]: done

File: config/routes.rb (hunk 7 above)
Comment (raptorf1) [reply config/routes.rb]: Thank you.
Updated ;)

File: features/visitor_can_navigate_the_site.feature (hunk 6 above)
Comment (raptorf1) [reply features/visitor_can_navigate_the_site.feature]: OK. Changed to 'When'. Edited the cucumber steps also.

File: features/step_definitions/basic_steps.rb (hunk 5 above)
Comment (raptorf1) [reply features/step_definitions/basic_steps.rb]: OK. Added ;)

File: features/step_definitions/basic_steps.rb (hunk 4 above)
Comment (raptorf1) [reply features/step_definitions/basic_steps.rb]: Also added.

File: features/visitor_can_navigate_the_site.feature (hunk 3 above)
Comment (raptorf1) [reply features/visitor_can_navigate_the_site.feature]: removed

Comment (raptorf1) [reply features/visitor_can_navigate_the_site.feature]: Removed

File: config/routes.rb (hunk 2 above)
Comment (raptorf1) [reply config/routes.rb]: OK. Rewrote them

File: app/controllers/about_us_controller.rb (hunk 8 above)
Comment (faraznaeem) [reply app/controllers/about_us_controller.rb]: You don't need a controller to navigate to different pages, you can reuse the products controller. 
https://stackoverflow.com/questions/18703349/how-to-add-a-new-action-to-the-existing-controller

That being said why do you start a story about navigating if there is nothing to navigate to?

**Review by faraznaeem — CHANGES_REQUESTED** [review]

File: app/controllers/about_us_controller.rb (hunk 8 above)
Comment (Carrosen) [reply app/controllers/about_us_controller.rb]: ahhh nice. I thought you did need one. That clears things up.

There is something to navigate to, to about us.

**Comment by raptorf1** [comment]
> Please add a screenshot as well

Screenshots added.

**Review by faraznaeem — APPROVED** [review]

File: app/controllers/products_controller.rb [hunk 10]
```diff
@@ -0,0 +1,8 @@
+class ProductsController < ApplicationController
+  def index
+  end
+
+  def about_us
+  end
```
Comment (tochman) [inline app/controllers/products_controller.rb:6]: 😳Why is this method defined?

File: config/routes.rb [hunk 11]
```diff
@@ -1,3 +1,6 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+
+  root controller: :products, action: :index
+  resources :products, only: [:index]
+  get "products/about_us"
```
Comment (tochman) [inline config/routes.rb:5]: Why is this route defined?

File: features/step_definitions/assertion_steps.rb [hunk 12]
```diff
@@ -0,0 +1,11 @@
...
+  visit root_path
+end
+
+When("I click {string} link") do |link|
+  click_link link
+end
+
+Then("I should see {string}") do |content|
+  expect(page).to have_content content 
+end
```
Comment (tochman) [inline features/step_definitions/assertion_steps.rb:11]: Assertions should be placed in a separate file

File: app/controllers/products_controller.rb (hunk 10 above)
Comment (raptorf1) [reply app/controllers/products_controller.rb:6]: To navigate to another page based on our design sprint. After Oliver's recommendation the action is removed.

File: config/routes.rb (hunk 11 above)
Comment (raptorf1) [reply config/routes.rb:5]: Look my reply above.

File: features/step_definitions/assertion_steps.rb (hunk 12 above)
Comment (raptorf1) [reply features/step_definitions/assertion_steps.rb:11]: You are in the assertion_steps.rb
